src/isArray.test.ts

```diff
@@ -2,6 +2,13 @@ import { isArray } from './isArray';
 import { typesDataProvider } from '../test/types_data_provider';
 
 describe('isArray', () => {
+  test('isArray: should infer ReadonlyArray<unknown> when given any', () => {
+    const data1 = {} as any;
+    if (isArray(data1)) {
+      const unknownArray = [] as ReadonlyArray<unknown>;
+      expectTypeOf(data1).toEqualTypeOf(unknownArray);
```

Comment:
Hi @eranhirsch ,

Thanks for the review!
This change causes the test to not fail when it should. I'm not sure why.
It's probably because we are testing with `any` itself.

I tried other changes besides this one here but they also don't work.

Do you have any other ideas?

Reply:
I don't understand what the issue is. This test shouldn't fail, the type you check here is exactly the type we expect it to be, isn't it?

Reply:
I understand now. Is the problem that it isn't refined from `any`? if that is the case you can add a negative test `expectTypeOf(data).not.toBeAny()`.

Tell me if that solves the problem.

Reply:
Yeah that was the problem.

I still had to do `expectTypeOf(data1[0]).toBeUnknown()` because `expectTypeOf(data1).toEqualTypeOf<ReadonlyArray<unknown>>` just does not seem to be safe and report anything when it should.

Feel free to change my code but these two `expectTypeOf` seem to cover what we wanted